src/core/server/integration_tests/saved_objects/migrations/zdt_2/sor_higher_version_docs.test.ts

```diff
@@ -138,6 +138,17 @@ describe('Higher version doc conversion', () => {
         newField: 'someValue',
       });
     });
+    it('throws error for documents using higher version model than current', async () => {
+      try {
+        await repositoryV1.get('test-type', 'doc-1', {
+          versionModelMatch: 'strict',
+        });
+      } catch (err) {
```

Comment:
👍

We could also add a test that asserts that retrieving the docs using the repositoryV2 doesn't throw even when `versionModelMatch: 'strict'` is used.